### ISSUE TITLE
additionalproperties no longer false

### ISSUE THREAD

**Issue by csMACnz** [domaindrivendev/Swashbuckle.AspNetCore#1625]
We were using version 5.0.0 of swashbuckle and it correctly (for our definition of correct) added the `additionalProperties: false` to our objects.  Upgrading beyond this version causes this to regress. 

Since the specification default for `additionalProperties` is true, clients will generate a dictionary to hold additional values which is not the behaviour we desire, and not the behaviour we had in 5.0.0

While I'm happy to accept that this is the new "correct", there isn't a workaround to get this value set to back to false for a given object. (in comparison, other libraries set this to false, and have a way to enable true, like the NSwag using [JsonSchemaExtensionDataAttribute](https://github.com/RicoSuter/NJsonSchema/wiki/JsonSchemaGenerator#provided-njsonschema-attributes))

This is stopping us upgrading to a newer version of swashbuckle at this stage, while we figure out a way to add these back in.

Let me know if I need to clarify any of the above in more detail.

Code where this "worked": https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/676c81c3370b675802fe6268b006eb0918f37836/src/Swashbuckle.AspNetCore.Newtonsoft/SchemaGenerator/NewtonsoftObjectHandler.cs#L50

PR where it was "removed": https://github.com/domaindrivendev/Swashbuckle.AspNetCore/commit/5c6e884ef538507d954f115108f7bb60bfaaec54

```
public class MyType
{
    public string Code { get; set; }

    public string Name { get; set; }

    public string Other { get; set; }
}
```

Before (5.0.0)
```yaml
MyType:
  type: object
  properties:
    code:
      type: string
      nullable: true
    name:
      type: string
      nullable: true
    other:
      type: string
      nullable: true
  additionalProperties: false
```

After (5.1.0-5.3.1)
```yaml
MyType:
  type: object
  properties:
    code:
      type: string
      nullable: true
    name:
      type: string
      nullable: true
    other:
      type: string
      nullable: true
```

**Comment by csMACnz** [comment]
As followup, this "fixes" the issue for me. 

```
internal class NoAdditionalProperties : ISchemaFilter
    {
        public void Apply(OpenApiSchema schema, SchemaFilterContext context)
        {
            if (schema.Type == "object")
            {
                schema.AdditionalPropertiesAllowed = false;
            }
        }
    }
```

Unsure if this now a requirement for me to do, or considered a true bug? Thanks!

**Comment by domaindrivendev** [comment]
This seems like a bug - I'm about to release a new version anyway, and will try get the fix for that included

**Comment by domaindrivendev** [comment]
Fixed by a9cb28f41a4c566f5c98e5a4c40bf8012b20b1bc

**Comment by domaindrivendev** [comment]
Available as of `v5.3.2`
